src/components/Inputs/Select.tsx

```diff
@@ -0,0 +1,65 @@
+import { forwardRef, useState, useCallback, type DetailedHTMLProps, type SelectHTMLAttributes } from "react";
+import { twJoin } from "tailwind-merge";
+import { RiArrowDownSLine } from "react-icons/ri";
+import { Portal } from "../Portal";
+import "./Select.css";
+
+export interface SelectOption {
+  value: string;
+  label: string;
+}
+
+export interface SelectProps
+  extends Omit<DetailedHTMLProps<SelectHTMLAttributes<HTMLDivElement>, HTMLDivElement>, "value" | "onChange"> {
+  options: SelectOption[];
+  value?: string;
```

Comment:
```suggestion
  open?: boolean;
  defaultOpen?: boolean;
  value?: string;
  defaultValue?: string;
```
lets follow MUI naming conventions. `isOpen` looks ugly

https://mui.com/material-ui/api/select/